### ISSUE TITLE
saved forms for filling stopped working: just shows toast "Warning!No saved custom login fields found."

### ISSUE THREAD

**Issue by succeedmr123** [keepassxreboot/keepassxc-browser#1601]
## Expected Behavior

Recognizing fields, rock-solid

## Current Behavior

KeepassXC Firefox addon does not fill any forms anymore, at all, with a false toast message info "Warning!No saved custom login fields found."

## Possible Solution

A bit more accountable / closer inspection than the average user shown there is an error as shared with the relevant console / debugger output. Now, if I can help to remove the showstopper for Linux system by testing fixes or compiling stuff and see how it behaves on my average Ubuntu-Studio  system for the benefit of all of us, I would be glad to help. Best

## Steps to Reproduce (for bugs)

1. Update Firefox to v99
2. Update KeepasssXC to latest KeePassXC - Version 2.7.1 Revision: 5916a8f
3. See how it all stops working and test connection over proxy (works)
4. Notice that form filling is broken - only toast message "Warning!No saved custom login fields found." Go to Dev tools and harvest Console and Debugger output. See it is related to DOM, and then extension error (while still not being an expert, trying to do a bit more useful report

## Debug info

Console:
...
:53.319 Error:  DOMException: Operation is not supported ui.js:335:17
09:15:56.559 Error:  DOMException: Operation is not supported ui.js:335:17
...
Debugger:
ExtensionContent.jsm file ->
...
[Line 310] document.addEventListener('mouseup', function(e) {
    if (!e.isTrusted) {
        return;
    }

    kpxcPasswordDialog.selected = null;
    kpxcDefine.selected = null;
    kpxcUI.mouseDown = false;
});

HTMLDivElement.prototype.appendMultiple = function(...args) {
    for (const a of args) {
        this.append(a);
    }
};

Element.prototype.getLowerCaseAttribute = function(attr) {
    return this.getAttribute(attr) ? this.getAttribute(attr).toLowerCase() : undefined;
};

Element.prototype._attachShadow = Element.prototype.attachShadow;
Element.prototype.attachShadow = function () {
    try {
        return this._attachShadow({ mode: 'closed' });
    } catch (e) {
        console.log('Error: ', e);
    }
};

Object.prototype.shadowSelector = function(value) {
    return this.shadowRoot ? this.shadowRoot.querySelector(value) : undefined;
};

Object.prototype.shadowSelectorAll = function(value) {
    return this.shadowRoot ? this.shadowRoot.querySelectorAll(value) : undefined;
};
[EOF]
---

KeePassXC - {VERSION} Version 2.7.1 Revision: 5916a8f
KeePassXC-Browser - {VERSION} 1.7.12 (07-04-2022)
Operating system:  Linux , Ubuntu Studio 20.04.4 LTS AMD64
Browser:  Firefox 99.0

**Comment by varjolintu** [comment]
What site are you using?

**Comment by succeedmr123** [comment]
@varjolintu 

tried more than one site, namely

learn.xamk.fi

typing.com

Both exhibit this behaviour. Testing sites alongside my production work time, am aware it might be site specific, will check Autodesk now;

Autodesk.com feedback:

1. failed with the difference of not having any toast message, no form filling, can not see anything in dev console.
2. Update: By coming back to Autodesk tab from this github writeup and clicking on extension icon, now it detected single-form site (but,it was already defined before as such.. like it lost previous preset). Okay, let us Test that context: I clicked to remember single form username only site
3. I try to refresh site and see what happens: simple refresh.. then ad filters off refresh with bypass browser cache .. Nope, still dead.
4.  Clicking icon and choosing to "choose custom login fields and refreshing" - Test Failed, nothing filled
5.  (As in the initial bug report - The database is apparently connected all the time over proxy, it just refuse to fill anything)

**Comment by varjolintu** [comment]
@succeedmr123 Thanks. I'll check it out.

**Comment by succeedmr123** [comment]
@varjolintu You are welcome, if necessary - feel free to suggest further tests on this Linux system

**Comment by varjolintu** [comment]
Having hard time to reproduce this one. I think I saw one similar error related to `appendMultiple()` prototype before but couldn't solve it before it disappeared.

**Comment by succeedmr123** [comment]
asof learn.xamk.fi -  just now, ~12 hours after the first tests, it apparently work as it should have.

What changed in the system: nothing fundamental, I may recall rebooting the machine (all apps are restarted), and enabling tmpfs for memcaching small app assigned objects that are usually written over /tmp dir  to storage media if tmp.mount is not enabled. (could have speedup in apps internal processing as a byproduct)

Having noticed that, let us try the other two sites then

typing.com

feedback: at first did not function, but no toast message error. good sign.
F5 Reloaded website, "Choose custom login fields" in addon icon menu, and specifically chosen separately user field, skipping psw and totp,

and for password field at the subsequent screens same tactic  - not choosing right adjacent field next to the psw field, skipping user and TOTP.

That worked.


Autodesk.com

acts weird in the following manner (list of moves to try get it running, and observations)

at first browser access request window from KeepassXC app (accepted)
not filling e-mail field
-did custom form select as username
after that filled user and swapped to other screen with password
-at that screen again could not recognize psw field

the way of makinfg a workaround around  that, like in handling typing.com login - failed here; 
 if I set at that screen custom password field, and skip user and TOTP like usual, on refresh it does not fill e-mail (i.e. user) field.

**Comment by varjolintu** [comment]
Some dynamically displayed fields are not detected correctly because some pages include an animation after DOM changes, which means the fields are there but because those are not visible at the moment (but only after the animation) those fields are ignored.
We've previously tried to add event listeners for animation end triggers but even those are not 100% reliable, and sometimes they cause more bad than good.
I'd suggest trying filling unidentified fields manually, especially after the 1.8.0 release is out, that has some changed concerning those fills.

**Comment by varjolintu** [comment]
Still having the issue with 1.8.1?

**Comment by lxg208** [comment]
I have the same problem with aa.com:

https://www.aa.com/loyalty/login?uri=%2floyalty%2flogin&previousPage=%2fhomePage.do&bookingPathStateId=&marketId=

KeePassXC-Browser Version: 1.8.2.2
KeePassXC Version: 2.7.1

**Comment by varjolintu** [comment]
@lxg208 I could reproduce this one. Thanks!

**Comment by irishquinn** [comment]
> autodesk.com

Does your browser password manager work on this page and / or dynamic fields? If so, set the KeePassXC plugin to disable itself on site's in which this problem is detected and use your browsers built in password manager on these sites. Or do what I do and use the post it note system, which is basically a post-it note that says "If KeePassXC Extension disabled on this page, then username is user321 and password is password with a capital P, lowercase a, uppercase SS, and the remaining characters are upper/lower in remaining order"  - I know it seems like a hassle, and insecure to use post it notes as a password manager, but I doubt anyone will read past the first couple of words and like, "get it".  Security through obscurity. Thanks for contributing.

**Comment by irishquinn** [comment]
> Still having the issue with 1.8.1?

Or he could use Chrome, this bug or whatever does not effect Chrome... @succeedmr123

**Comment by varjolintu** [comment]
@irishquinn For `autodesk.com`, open your database, then the extension popup from browser's toolbar icon and enable the suggested Username Only option. In the future, please make a new issue for different problems. We are having a lot of "issue spam" where a thread can actually contain 2-3 different issues.
